### ISSUE TITLE
Considering using a learning rate scheduler for the trainings

### ISSUE THREAD

**Issue by abelsalm** [ivadomed/lumbar-classification-rsna-challenge-2024#43]
I know @tomDag25 and I simply runned training using a constant learning rate for now.

However PyTorch has [learning rate schedulers](https://pytorch.org/docs/stable/optim.html) to adapt learning rate during the training. I'm considering using those to test MIL models and I think we should do this systematically since those functions can be really helpful and easy to use.

I'm especially thinking of using this one : [CosineAnnealingLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CosineAnnealingLR.html)
It seems prettty adapted to our problem where models tend to overfitt quickly : it adapts lr from a maximum to a minimum using a cosine function, and can be programmed to only decrease during training (half cosine period), or going through multiple cosine  periods to avoid local optimums (thinking of NFN issues for instance here...)

**Comment by tomDag25** [comment]
This has been implemented for MIL training
